File: DuDoong-Domain/src/main/java/band/gosrock/domain/domains/host/repository/HostRepository.java
package band.gosrock.domain.domains.host.repository;


import band.gosrock.domain.domains.host.domain.Host;
import java.util.List;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.repository.CrudRepository;

public interface HostRepository extends CrudRepository<Host, Long>, HostCustomRepository {
    List<Host> findAllByMasterUserId(Long userId);

    List<Host> findAllByHostUsers_UserId(Long userId);

    Page<Host> findAllByHostUsers_UserId(Long userId, Pageable pageable);
}

File: DuDoong-Domain/src/main/kotlin/band/gosrock/domain/domains/host/repository/HostRepository.kt
package band.gosrock.domain.domains.host.repository

import band.gosrock.domain.domains.host.domain.Host
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.data.repository.CrudRepository

interface HostRepository : CrudRepository<Host, Long>, HostCustomRepository {
    fun findAllByMasterUserId(userId: Long): List<Host>
    fun findAllByHostUsers_UserId(userId: Long): List<Host>
    fun findAllByHostUsers_UserId(userId: Long, pageable: Pageable): Page<Host>
    fun findByHostUsersIdIn(userId: List<Long>): List<Host>
}
